src/bch_validator.js

```diff
@@ -0,0 +1,48 @@
+var cryptoUtils = require('./crypto/utils');
+var bech32 = require('./crypto/bech32');
+var BTCValidator = require('./bitcoin_validator');
+
+function validateAddress(address, currency, opts) {
+    var networkType = opts ? opts.networkType : ''
+    var prefix = 'bitcoincash';
+    var regexp = new RegExp(currency.regexp);
+    var raw_address;
+
+    var res = address.split(':');
+    if (res.length === 1) {
+        raw_address = address
+    } else {
+        if (res[0] !== 'bitcoincash') {
+            return false;
+        }
+        raw_address = res[1];
+    }
+
+    if (!regexp.test(raw_address)) {
+        return false;
+    }
+
+    if (raw_address.toLowerCase() != raw_address && raw_address.toUpperCase() != raw_address) {
```

Comment:
not sure if these `!=` are intended as they support coercion and the non-coercion version is used below

```suggestion
    if (raw_address.toLowerCase() !== raw_address && raw_address.toUpperCase() !== raw_address) {
```